portable/Zynq.2019.3/ff_sddisk.c

```diff
@@ -407,7 +407,7 @@ static CacheMemoryInfo_t * pucGetSDIOCacheMemory( BaseType_t xPartition )
 
     if( ( xPartition < 0 ) || ( xPartition >= ffconfigMAX_PARTITIONS ) )
     {
-        FF_PRINTF( "pucGetSDIOCacheMemory: bad partition number: %d ( max %d )\n", xPartition, ffconfigMAX_PARTITIONS - 1 );
+        FF_PRINTF( "pucGetSDIOCacheMemory: bad partition number: %ld ( max %d )\n", xPartition, ffconfigMAX_PARTITIONS - 1 );
```

Comment:
I tend not to use the 'l' (long) prefix as in '%ld' and '%lu', because it will cause problems on other platforms.
In stead I propose to add the following casts:
~~~diff
FF_PRINTF( "pucGetSDIOCacheMemory: bad partition number: %d ( max %d )\n",
-           xPartition,
+           ( int ) xPartition,
-           ffconfigMAX_PARTITIONS - 1 );
+           ( int ) (ffconfigMAX_PARTITIONS - 1 );
~~~
That works on any platform.
The same applies to all other proposed changes to the `FF_PRINTF()` formats.